internal/pkg/template/workload.go

```diff
@@ -540,14 +540,54 @@ type DeadLetterQueue struct {
 
 // NetworkOpts holds AWS networking configuration for the workloads.
 type NetworkOpts struct {
-	SecurityGroups []string
+	SecurityGroups []SecurityGroup
 	AssignPublicIP string
 	// SubnetsType and SubnetIDs are mutually exclusive. They won't be set together.
 	SubnetsType              string
 	SubnetIDs                []string
 	DenyDefaultSecurityGroup bool
 }
 
+// SecurityGroup represents the ID of an additional security group associated with the tasks.
+type SecurityGroup interface {
+	importable
+	Value() string
+}
+
+// PlainSecurityGroup returns a SecurityGroup that is a plain string value.
+func PlainSecurityGroup(value string) Variable {
```

Comment:
Why don't we make `plainSecurityGroup` public as there's no getter/setter in go 🤔  Also `plainSecurityGroup` implementing `Variable` interface which "represents the value of an environment variable" looks weird to me...

Reply:
I feel like maybe `Variable` is not a generic enough name for what methods (`RequiresImport()`, `Value()`) the interface defines. It sounds more like a concrete struct name.

Reply:
Good catch! it is a mistake from my end the function is supposed to be
`func PlainSecurityGroup(value string) SecurityGroup` which is implementing `SecurityGroup` interface.

SecurityGroup interface looks something like this
```
type SecurityGroup interface {
	importable
	Value() string
}
```
 I will make `plainSecurityGroup` and `importedSecurityGroup` public and remove these setter functions
`1. func PlainSecurityGroup(value string) SecurityGroup` 
`2. func ImportedSecurityGroup(name string) SecurityGroup`

Reply:
Hmm it doesn't sound necessary tho because both `SecurityGroup` and `Variable` interface will be used only once (aka not sharable and generic enough to justify their existence). I would suggest we **either**

1. [Recommended] Rename `Variable` interface to a more generic name (e.g., `Importable` or a better name) reflecting what methods it contains
2. Remove interfaces and use structs only

Reply:
Done.